border-image repaint code is broken in some writing modes

smfr opened the pull request:
#### e90b24b974c2c372167cef18b3d2ea68bcc774f0
<pre>
border-image repaint code is broken in some writing modes
<a href="https://bugs.webkit.org/show_bug.cgi?id=293552">https://bugs.webkit.org/show_bug.cgi?id=293552</a>
<a href="https://rdar.apple.com/152396671">rdar://152396671</a>

Reviewed by Alan Baradlay.

Similarly to 295977@main, this fixes repainting with border-image-outsets in some writing
modes; we have to flip the outsets into the overflow-based coordinate system.

Tidy up the fix from 295977@main, removing `shadowOutsetsForWritingMode()` and adding
`convertOutsetsToOverflowCoordinates()`.

* LayoutTests/fast/repaint/border-image-repaint-sideways-writing-modes-expected.txt: Added.
* LayoutTests/fast/repaint/border-image-repaint-sideways-writing-modes.html: Added.
* LayoutTests/fast/repaint/border-image-repaint-writing-modes-expected.txt: Added.
* LayoutTests/fast/repaint/border-image-repaint-writing-modes.html: Added.
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::convertOutsetsToOverflowCoordinates):
(WebCore::RenderBox::applyVisualEffectOverflow const):
(WebCore::shadowOutsetsForWritingMode): Deleted.

Canonical link: <a href="https://commits.webkit.org/296016@main">https://commits.webkit.org/296016@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/815ae82b7f5a2fee12e12613efd01de3d3020019

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106946 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26653 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17050 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112154 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57506 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/108939 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27326 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35155 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81208 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109904 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21695 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96465 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61553 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21134 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/14575 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56952 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91037 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14602 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/115180 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34039 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25113 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90261 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34407 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92696 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89976 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22960 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34893 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12708 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/29753 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33963 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/39418 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33711 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37063 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35343 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9cad586ec7aedafbb040d458a239d635dd001f29. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46502

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9cad586ec7aedafbb040d458a239d635dd001f29. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46502

smfr committed:
border-image repaint code is broken in some writing modes
https://bugs.webkit.org/show_bug.cgi?id=293552
rdar://152396671

Reviewed by Alan Baradlay.

Similarly to 295977@main, this fixes repainting with border-image-outsets in some writing
modes; we have to flip the outsets into the overflow-based coordinate system.

Tidy up the fix from 295977@main, removing `shadowOutsetsForWritingMode()` and adding
`convertOutsetsToOverflowCoordinates()`.

* LayoutTests/fast/repaint/border-image-repaint-sideways-writing-modes-expected.txt: Added.
* LayoutTests/fast/repaint/border-image-repaint-sideways-writing-modes.html: Added.
* LayoutTests/fast/repaint/border-image-repaint-writing-modes-expected.txt: Added.
* LayoutTests/fast/repaint/border-image-repaint-writing-modes.html: Added.
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::convertOutsetsToOverflowCoordinates):
(WebCore::RenderBox::applyVisualEffectOverflow const):
(WebCore::shadowOutsetsForWritingMode): Deleted.

Canonical link: https://commits.webkit.org/296016@main
